### PR TITLE
chore: adopt API v7 - WPB-12024

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-12024" title="WPB-12024" target="_blank">TaskWPB-12024</a>  [iOS] Adopt API v7
  </td></table>
  <br />
 

### Issue

This PR enables version 7 of the api and applies the changes.

### Testing

Test all features of the app against a v7 backend.

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios-sync-engine/Tests/Source/Synchronization/Transcoders/UserProfileUpdateRequestStrategyTests.swift [hunk 1]
```diff
@@ -159,34 +160,35 @@ extension UserProfileUpdateRequestStrategyTests {
...
+            // GIVEN
+            userProfileUpdateStatus.suggestHandles()
+            XCTAssertTrue(waitForAllGroupsToBeEmpty(withTimeout: 0.2), "failed for APIVersion.v\(apiVersion.rawValue)")
+            guard let handles = userProfileUpdateStatus.suggestedHandlesToCheck else {
+                XCTFail("failed for APIVersion.v\(apiVersion.rawValue)")
+                continue
+            }
+
+            // WHEN
+            let possibleRequest = sut.nextRequest(for: apiVersion)
```
Comment (caldrian) [inline wire-ios-sync-engine/Tests/Source/Synchronization/Transcoders/UserProfileUpdateRequestStrategyTests.swift]: I tried to run this test for all API versions, but failed and reverted it.

File: WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV5.swift [hunk 2]
```diff
@@ -55,7 +59,7 @@ class ConversationsAPIV5: ConversationsAPIV4 {
             throw ConversationsAPIError.userAndDomainShouldNotBeEmpty
         }
 
-        let components = URLComponents(string: "\(pathPrefix)\(basePath)/one2one/\(domain)/\(userID)")
+        let components = URLComponents(string: "\(oneToOneConversationsPath)/\(domain)/\(userID)")
```
Comment (caldrian) [inline WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV5.swift:62]: I wanted to fix this path, but then I had to make all the other changes as well (extending APIVersion etc.).

File: WireAPI/Sources/WireAPI/Models/API/APIVersion.swift [hunk 3]
```diff
@@ -42,7 +43,7 @@ public enum APIVersion: UInt, CaseIterable, Comparable {
     /// Only if these critera are met should we explicitly mark the version
     /// as production ready.
 
-    public static let productionVersions: Set<Self> = [.v0, .v1, .v2, .v3, .v4, .v5]
+    public static let productionVersions: Set<Self> = [.v0, .v1, .v2, .v3, .v4, .v5, .v6, .v7]
```
Comment (netbe) [inline WireAPI/Sources/WireAPI/Models/API/APIVersion.swift]: maybe we should put it in development first then in prod, or is it already finalised?

File: WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV7.swift [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+final class ConversationsAPIV7: ConversationsAPIV6 {
+
+    override var apiVersion: APIVersion { .v7 }
+
+    override var oneToOneConversationsPath: String {
```
Comment (netbe) [inline WireAPI/Sources/WireAPI/APIs/ConversationsAPI/ConversationsAPIV7.swift:23]: renamed POST /conversations/one2one → POST /one2one-conversations

renamed GET /conversations/one2one → GET /one2one-conversations

File: wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/SyncMLSOneToOneConversationActionHandler.swift [hunk 5]
```diff
@@ -47,8 +47,11 @@ final class SyncMLSOneToOneConversationActionHandler: ActionHandler<SyncMLSOneTo
             return nil
         }
 
+        let path = apiVersion >= .v7
```
Comment (netbe) [inline wire-ios-request-strategy/Sources/Request Strategies/MLS/Actions/SyncMLSOneToOneConversationActionHandler.swift:50]: renamed GET /conversations/one2one → GET /one2one-conversations

File: wire-ios-sync-engine/Source/Data Model/ServiceUser.swift [hunk 6]
```diff
@@ -113,7 +113,9 @@ private extension ServiceUserData {
             fatal("conversation is not synced with the backend")
         }
 
-        let path = "/conversations/\(remoteIdentifier.transportString())/bots"
+        let path = apiVersion >= .v7
```
Comment (netbe) [inline wire-ios-sync-engine/Source/Data Model/ServiceUser.swift:116]: renamed POST /conversations/bots → POST /bot/conversations

File: wire-ios-sync-engine/Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift [hunk 7]
```diff
@@ -132,8 +132,11 @@ public class UserProfileUpdateRequestStrategy: AbstractRequestStrategy, ZMSingle
 
         case handleCheckSync:
             let handle = userProfileUpdateStatus.handleToCheck!
+            let path = apiVersion >= .v7
```
Comment (netbe) [inline wire-ios-sync-engine/Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift:135]: renamed HEAD /users/handles → HEAD /handles

File: wire-ios-sync-engine/Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift [hunk 8]
```diff
@@ -152,12 +155,15 @@ public class UserProfileUpdateRequestStrategy: AbstractRequestStrategy, ZMSingle
             guard let handlesToCheck = userProfileUpdateStatus.suggestedHandlesToCheck else {
                 fatal("Tried to check handles availability, but no handle was available")
             }
+            let path = apiVersion >= .v7
```
Comment (netbe) [inline wire-ios-sync-engine/Source/Synchronization/Strategies/UserProfileUpdateRequestStrategy.swift:158]: renamed POST /users/handles → POST /handles

File: wire-ios-sync-engine/Tests/Source/Synchronization/Transcoders/ZMUserImageTranscoderTests.m [hunk 9]
```diff
@@ -1,791 +0,0 @@
...
-        XCTAssertEqual(request.acceptedResponseMediaTypes, ZMTransportAcceptImage);
-        XCTAssertNil(request.payload);
-    }];
-}
-
-- (void)testThatItUdpatesTheMediumImageFromAResponse;
-{
-    // TODO: There's a race condition here.
-    // If the mediumRemoteIdentifier changes while we're retrieving image data
-    // we'll still assume that the data is the newest. In order to fix that,
```
Comment (netbe) [inline wire-ios-sync-engine/Tests/Source/Synchronization/Transcoders/ZMUserImageTranscoderTests.m:402]: unused code ?

File: wire-ios-transport/Source/Public/APIVersion.swift [hunk 10]
```diff
@@ -49,11 +48,6 @@ extension APIVersion: Comparable {
 
 public extension APIVersion {
     var useQualifiedIds: Bool {
-        switch self {
-        case .v0:
-            false
-        case .v1, .v2, .v3, .v4, .v5, .v6:
-            true
-        }
+        self != .v0
```
Comment (netbe) [inline wire-ios-transport/Source/Public/APIVersion.swift:51]: nice

**Review by netbe — COMMENTED** [review]
I left a couple of comments.

I wonder also if we need to update the capabilities on WIreAPI for getSelfClient.

File: wire-ios-sync-engine/Tests/Source/Synchronization/Transcoders/ZMUserImageTranscoderTests.m (hunk 9 above)
Comment (caldrian) [reply wire-ios-sync-engine/Tests/Source/Synchronization/Transcoders/ZMUserImageTranscoderTests.m:402]: Sorry, I should have left a comment.
Yes, this file wasn't referenced in the Xcode project. 
It seems to have been abandoned already before moving to the monorepo.

File: WireAPI/Sources/WireAPI/Models/API/APIVersion.swift (hunk 3 above)
Comment (caldrian) [reply WireAPI/Sources/WireAPI/Models/API/APIVersion.swift]: I'm not sure to be honest. @johnxnguyen do you know the answer?

Comment (johnxnguyen) [reply WireAPI/Sources/WireAPI/Models/API/APIVersion.swift]: the sequence of steps should be:

1. backend freezes api v7
2. client implement v7, but leave it as a development version
3. QA runs full automation with v7 enabled
4. If all checks pass, client moves v7 to production version.

**Review by johnxnguyen — CHANGES_REQUESTED** [review]
Looks good, but I would keep version in development until QA runs automation.

File: WireAPI/Sources/WireAPI/Models/API/APIVersion.swift (hunk 3 above)
Comment (caldrian) [reply WireAPI/Sources/WireAPI/Models/API/APIVersion.swift]: Fixed, thanks! 87c4d034a41c1203255d0e5c322889a0b35a8160

**Review by johnxnguyen — APPROVED** [review]

**Review by netbe — APPROVED** [review]
LGTM :)

**Pull request merged**